bitnami/schema-registry/templates/statefulset.yaml

```
@@ -6,8 +6,12 @@ SPDX-License-Identifier: APACHE-2.0
 {{- $fullname := include "common.names.fullname" .  }}
 {{- $releaseNamespace := .Release.Namespace }}
 {{- $clusterDomain := .Values.clusterDomain }}
-{{- $kafkaProtocol := include "schema-registry.listenerType" ( dict "protocol" (ternary .Values.kafka.auth.clientProtocol .Values.externalKafka.auth.protocol .Values.kafka.enabled) ) -}}
-{{- $kafkaReplicaCount := int .Values.kafka.replicaCount }}
+{{- $kafkaProtocol := upper (ternary .Values.kafka.listeners.client.protocol .Values.externalKafka.listener.protocol .Values.kafka.enabled) }}
+{{- $kafkaBrokerReplicaCount := int .Values.kafka.broker.replicaCount }}
+{{- $kafkaControllerReplicaCount := int .Values.kafka.controller.replicaCount }}
+{{- if or (not .Values.kafka.kraft.enabled) (.Values.kafka.controller.controllerOnly) }}
+{{- $kafkaControllerReplicaCount = 0 }}
+{{- end }}
```

Not sure if this is done in purpose but using `ternary` should be viable here right?

```suggestion
{{- $kafkaControllerReplicaCount := ternary 0 (int .Values.kafka.broker.replicaCount) (or (not .Values.kafka.kraft.enabled) (.Values.kafka.controller.controllerOnly)) }}
```